auctions/dutch-auction/src/main.sw

```diff
@@ -0,0 +1,280 @@
+contract;
+
+dep abi;
+dep data_structures;
+dep errors;
+dep events;
+
+use std::{
+    address::Address,
+    assert::require,
+    block::height,
+    chain::auth::{AuthError, msg_sender},
+    context::{call_frames::msg_asset_id, msg_amount},
+    contract_id::ContractId,
+    identity::Identity,
+    logging::log,
+    result::*,
+    revert::revert,
+    storage::StorageMap,
+    token::{force_transfer_to_contract, transfer_to_output}
+};
+
+use abi::DutchAuction;
+use data_structures::Auction;
+use errors::{SetupError, TimeError, UserError};
+use events::{CancelledAuctionEvent, CreatedAuctionEvent, WinningBidEvent};
+
+storage {
+    /// Mapping an auction_id to its respective auction, allowing for multiple auctions to happen simultaneously
+    auctions: StorageMap<u64,
+    Auction>, /// Tracking how many auctions have been made till now
+    auction_count: u64,
+    /// Auction ids of the active auctions per beneficiary
+    active_auctions_by_identity: StorageMap<Identity,
```

Comment:
I would rename this and split this into multiple mappings so that the UI can make a query for each ID. I'll show you the implementation when it hits the fundraiser draft

Reply:
for the data of a specific auction they can just use the auction function and enter the id in there?

Reply:
Take a look at the fundraiser draft that I've mentioned

Reply:
I did, still dont see why someone cant just do `auction(id).author` instead

Reply:
what? don't use arrays. they are static. Look at the fundraiser again and use hashing / indexing to retrieve data

Reply:
Have you read the fundraiser draft PR?

Reply:
Yes and after seeing it once again im more confused than before. Please specify which implementation exactly you are talking about.

> so that the UI can make a query for each ID.
i can only imagine you are referring to the auction id here, there is already a function to get the details of one specific auction